### PR TITLE
Odata feeds: Add Repeat Group, Parent Cases, and De-Id support

### PR DESCRIPTION
**Pull request by biyeun**
Adds support for Repeat Groups, Parent Cases, and De-Identifying data in odata feeds.

Also adds support for expanding checkbox questions / split columns.

Here is what the UI looks like for additional feed support for repeats / parent cases. Dev suggested the pop up in case there's a lot of repeat groups or multiple parent cases:

Screen Shot 2019-08-08 at 1 19 28 PM
Screen Shot 2019-08-08 at 1 19 11 PM

Screen Shot 2019-08-08 at 1 19 34 PM
Screen Shot 2019-08-08 at 1 19 45 PM

cc @dimagi/product

### CODE REVIEW

File: corehq/apps/api/urls.py [hunk 1]
```diff
@@ -97,9 +97,14 @@ def top_level(self, request, api_name=None, **kwargs):
...
 def api_url_patterns():
     # todo: these have to come first to short-circuit tastypie's matching
+    yield url(r'v0.5/odata/cases/(?P<config_id>[\w\-:]+)/(?P<table_id>[\d]+)/$', ODataCaseServiceView.as_view(), name=ODataCaseServiceView.urlname_table)
     yield url(r'v0.5/odata/cases/(?P<config_id>[\w\-:]+)/$', ODataCaseServiceView.as_view(), name=ODataCaseServiceView.urlname)
+    yield url(r'v0.5/odata/cases/(?P<config_id>[\w\-:]+)/(?P<table_id>[\d]+)/\$metadata$', ODataCaseMetadataView.as_view(), name=ODataCaseMetadataView.urlname_table)
     yield url(r'v0.5/odata/cases/(?P<config_id>[\w\-:]+)/\$metadata$', ODataCaseMetadataView.as_view(), name=ODataCaseMetadataView.urlname)
+
+    yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/(?P<table_id>[\d]+)/$', ODataFormServiceView.as_view(), name=ODataFormServiceView.urlname_table)
     yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/$', ODataFormServiceView.as_view(), name=ODataFormServiceView.urlname)
+    yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/(?P<table_id>[\d]+)/\$metadata$', ODataFormMetadataView.as_view(), name=ODataFormMetadataView.urlname_table)
```
[the remark this answers is not in the document]
Comment (biyeun) [reply corehq/apps/api/urls.py]: ignoring

File: corehq/apps/api/urls.py [hunk 2]
```diff
@@ -97,9 +97,14 @@ def top_level(self, request, api_name=None, **kwargs):
 
 def api_url_patterns():
     # todo: these have to come first to short-circuit tastypie's matching
+    yield url(r'v0.5/odata/cases/(?P<config_id>[\w\-:]+)/(?P<table_id>[\d]+)/$', ODataCaseServiceView.as_view(), name=ODataCaseServiceView.urlname_table)
     yield url(r'v0.5/odata/cases/(?P<config_id>[\w\-:]+)/$', ODataCaseServiceView.as_view(), name=ODataCaseServiceView.urlname)
+    yield url(r'v0.5/odata/cases/(?P<config_id>[\w\-:]+)/(?P<table_id>[\d]+)/\$metadata$', ODataCaseMetadataView.as_view(), name=ODataCaseMetadataView.urlname_table)
     yield url(r'v0.5/odata/cases/(?P<config_id>[\w\-:]+)/\$metadata$', ODataCaseMetadataView.as_view(), name=ODataCaseMetadataView.urlname)
+
+    yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/(?P<table_id>[\d]+)/$', ODataFormServiceView.as_view(), name=ODataFormServiceView.urlname_table)
```
[the remark this answers is not in the document]
Comment (biyeun) [reply corehq/apps/api/urls.py]: ignoring

File: corehq/apps/api/odata/serializers.py [hunk 3]
```diff
@@ -79,10 +98,19 @@ def to_json(self, data, options=None):
         domain = data.pop('domain', None)
         config_id = data.pop('config_id', None)
         api_path = data.pop('api_path', None)
-        assert all([domain, config_id, api_path]), [domain, config_id, api_path]
+        table_id = data.pop('table_id', None)
+
+        assert (all([domain, config_id, api_path, table_id]),
```
[the remark this answers is not in the document]
Comment (biyeun) [reply corehq/apps/api/odata/serializers.py]: ignoring

File: corehq/apps/api/odata/serializers.py [hunk 4]
```diff
@@ -27,10 +48,19 @@ def to_json(self, data, options=None):
...
```
[the remark this answers is not in the document]
Comment (biyeun) [reply corehq/apps/api/odata/serializers.py]: ignoring

**Comment by czue** [comment]
didn't look at code but feature is :fire:

**Review by esoergel — APPROVED** [review]

File: corehq/apps/api/odata/serializers.py [hunk 5]
```diff
@@ -27,10 +27,19 @@ def to_json(self, data, options=None):
         domain = data.pop('domain', None)
         config_id = data.pop('config_id', None)
         api_path = data.pop('api_path', None)
-        assert all([domain, config_id, api_path]), [domain, config_id, api_path]
+        table_id = data.pop('table_id', None)
```
[the remark this answers is not in the document]
Comment (biyeun) [reply corehq/apps/api/odata/serializers.py]: yeah...i know. there's a lot of duplicated code in the APIs for this. I'm not sure if that was intentional on nick's part (to make sure no APIs were tightly coupled with each other) or just a speed thing...

File: corehq/apps/api/odata/serializers.py [hunk 6]
```diff
@@ -27,10 +48,19 @@ def to_json(self, data, options=None):
...
+
+        assert (all([domain, config_id, api_path, table_id]),
+                [domain, config_id, api_path, table_id])
```
[the remark this answers is not in the document]
Comment (biyeun) [reply corehq/apps/api/odata/serializers.py]: gotcha. ok i'll fix

File: corehq/apps/api/odata/serializers.py [hunk 7]
```diff
@@ -16,28 +16,13 @@
...
-        rows = table.get_rows(
-            document,
-            row_number,
-            split_columns=config.split_multiselects,
-            transform_dates=config.transform_dates,
-            as_json=True,
-        )
-        data.extend(rows)
-    return data
+class ODataBaseSerializer(Serializer):
```
Comment (esoergel) [inline corehq/apps/api/odata/serializers.py]: :star:

**Review by esoergel — APPROVED** [review]
Great stuff!

**Comment by djmore9** [comment]
@biyeun crushing it as usual!

**Pull request merged**